src/ipc/ipc3/helper.c

```diff
@@ -278,11 +278,7 @@ static void comp_specific_builder(struct sof_ipc_comp *comp,
 	case SOF_COMP_NONE:
 		config->process.type = proc->type;
 		config->process.size = proc->size;
-#if CONFIG_LIBRARY || UNIT_TEST
-		config->process.data = proc->data + comp->ext_data_length;
-#else
-		config->process.data = proc->data;
-#endif
+		config->process.data = NULL;
```

Comment:
no, `proc->data` is an array, so this is perfectly valid

Reply:
Why do we handle this differently on host ? Expectation is that ABI should be the same, I ssuspect the testbench is doing something different @singalsu @andrula-song ?

Reply:
I think the data advance to past possible UUID can be done in every build, not just UT and testbench.